File: src/main/java/io/github/oleksivio/telegram/bot/core/controller/annotations/parser/unpacker/ChannelPostUnpackerParser.java
package io.github.oleksivio.telegram.bot.core.controller.annotations.parser.unpacker;

import org.springframework.stereotype.Component;
import io.github.oleksivio.telegram.bot.api.annotations.receiver.ChannelPostReceiver;
import io.github.oleksivio.telegram.bot.api.model.objects.std.Message;
import io.github.oleksivio.telegram.bot.core.controller.handler.unpack.UpdateUnpacker;

import java.util.Optional;

@Component
public class ChannelPostUnpackerParser extends UnpackerParser<ChannelPostReceiver, Message> {

    @Override
    public Class<ChannelPostReceiver> getAnnotationClass() {
        return ChannelPostReceiver.class;
    }

    @Override
    public UpdateUnpacker<Message> getUpdateUnpacker() {
        return new UpdateUnpacker<>(in -> Optional.ofNullable(in.getChannelPost()), Message.class);
    }

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/core/controller/annotations/parser/unpacker/ChannelPostUnpackerParser.kt
package io.github.oleksivio.telegram.bot.core.controller.annotations.parser.unpacker

import io.github.oleksivio.telegram.bot.api.annotations.receiver.ChannelPostReceiver
import io.github.oleksivio.tl.kbot.server.api.objects.std.Message
import io.github.oleksivio.telegram.bot.core.controller.handler.unpack.UpdateUnpacker
import org.springframework.stereotype.Component

@Component
class ChannelPostUnpackerParser : UnpackerParser<ChannelPostReceiver, Message>() {
    override val updateUnpacker: UpdateUnpacker<Message>
        get() = UpdateUnpacker({ it.channelPost }, Message::class)

    override val parserAnnotationClass = ChannelPostReceiver::class

}



